flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/GroupReduceITCase.java

```diff
@@ -1063,6 +1065,33 @@ public void reduce(Iterable<Tuple5<Integer, Long, Integer, String, Long>> values
 
 	}
 
+	@Test
+	public void testGroupReduceWithAtomicValue() throws Exception {
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		DataSet<Integer> ds = env.fromElements(1, 1, 2, 3, 4);
+		DataSet<Integer> reduceDs = ds.groupBy("*").reduceGroup(new GroupReduceFunction<Integer, Integer>() {
+			@Override
+			public void reduce(Iterable<Integer> values, Collector<Integer> out) throws Exception {
+				Set<Integer> set = new HashSet<Integer>();
```

Comment:
Since all elements of a group are identical when you do `groupBy("*")` this could be implemented a bit easier by just returning the first element of a group (`out.collect(values.iterator().next())`).